src/bootstrap/src/core/build_steps/test.rs

```diff
@@ -1344,6 +1353,61 @@ macro_rules! coverage_test_alias {
     };
 }
 
+#[derive(Debug, Copy, Clone, Hash, PartialEq, Eq, Ord, PartialOrd)]
+pub struct RunMakeSupport {
+    pub compiler: Compiler,
+    pub target: TargetSelection,
+}
+
+impl Step for RunMakeSupport {
+    type Output = PathBuf;
+    const DEFAULT: bool = true;
+
+    fn should_run(run: ShouldRun<'_>) -> ShouldRun<'_> {
+        run.never()
+    }
+
+    fn run(self, builder: &Builder<'_>) -> PathBuf {
+        let cargo = tool::prepare_tool_cargo(
+            builder,
+            self.compiler,
+            Mode::ToolStd,
+            self.target,
+            "build",
+            "src/tools/run-make-support",
+            SourceType::InTree,
+            &[],
+        );
+
+        let mut cargo = Command::from(cargo);
+
+        let mut stage1_std_dir =
+            builder.stage_out(builder.compiler(1, builder.config.build), Mode::Std);
+        stage1_std_dir.push(self.target.triple);
+        stage1_std_dir.push("release");
```

Comment:
Is it possible to build RunMakeSupport with the bootstrap compiler instead? For cg_clif I'm using `./x.py test --stage 0` to avoid unnecessarily building the stage1 compiler as I'm using the nightly matching to the rustc source as bootstrap compiler. As far as I can tell this PR would build stage 1 unconditionally when doing a run-make test. Also I believe you are using the wrong rustc to build the rmake.rs files when testing stage 2.

Reply:
I don't think I got `RunMakeSupport` to work with the bootstrap compiler. I tried to at least respect the stages in 0c090b7, but I'm not very confident my changes are correct.

Reply:
I think the current code to respect stages is fine.